lingua_franca/lang/parse_en.py

```diff
@@ -1685,3 +1710,256 @@ def numbers_to_digits(self, utterance):
+            elif unit == "microseconds":
+                if relative_units["microseconds"] is None:
+                    relative_units["microseconds"] = 0
+                relative_units["microseconds"] += value
+            # relativedelta does not support decades, centuries or millennia
+            elif unit == "years":
+                if relative_units["years"] is None:
+                    relative_units["years"] = 0
+                relative_units["years"] += value
+            elif unit == "decades":
+                if relative_units["years"] is None:
+                    relative_units["years"] = 0
+                relative_units["years"] += value * 10
+            elif unit == "centurys":
+                if relative_units["years"] is None:
+                    relative_units["years"] = 0
+                relative_units["years"] += value * 100
+            elif unit == "millenniums":
+                if relative_units["years"] is None:
+                    relative_units["years"] = 0
+                relative_units["years"] += value * 1000
+            else:
+                relative_units[unit] = value
+
+        # microsecond, month, year must be ints
+        relative_units["microseconds"] = int(relative_units["microseconds"])
+        if time_unit == TimespanUnit.RELATIVEDELTA_FALLBACK:
+            for unit in ["months", "years"]:
+                value = relative_units[unit]
+                _leftover, _ = math.modf(value)
+                if _leftover != 0:
+                    print("[WARNING] relativedelta requires {unit} to be an "
+                          "integer".format(unit=unit))
+                    # fallback to timedelta resolution
+                    return extract_timespan_en(original_text,
+                                               TimespanUnit.TIMEDELTA,
+                                               replace_token)
+                relative_units[unit] = int(value)
+        elif time_unit == TimespanUnit.RELATIVEDELTA_APPROXIMATE:
+            _leftover, year = math.modf(relative_units["years"])
+            relative_units["months"] += 12 * _leftover
+            relative_units["years"] = int(year)
+            _leftover, month = math.modf(relative_units["months"])
+            relative_units["days"] += DAYS_IN_1_MONTH * _leftover
+            relative_units["months"] = int(month)
+        else:
+            for unit in ["months", "years"]:
+                value = relative_units[unit]
+                _leftover, _ = math.modf(value)
+                if _leftover != 0:
+                    raise ValueError("relativedelta requires {unit} to be an "
+                                     "integer".format(unit=unit))
+                relative_units[unit] = int(value)
+        duration = relativedelta(**relative_units) if \
+            any(relative_units.values()) else None
+    else:
+        microseconds = 0
+        units = ['months', 'years', 'decades', 'centurys', 'millenniums',
```

Comment:
Same as above, `centuries` and `millenia`. Maybe these units should be in some global?

Reply:
this is just an hack because of logic below stripping an s, to avoid the need for more code and reuse the existing loop

i suppose a comment should be added for clarification